libPostTagSystem/PostTagSearcher.hpp

```diff
@@ -0,0 +1,81 @@
+#ifndef LIBPOSTTAGSYSTEM_POSTTAGSEARCHER_HPP_
+#define LIBPOSTTAGSYSTEM_POSTTAGSEARCHER_HPP_
+
+#include <memory>
+#include <vector>
+
+#include "TagState.hpp"
+
+namespace PostTagSystem {
+class PostTagSearcher {
+ public:
+  PostTagSearcher();
+
+  enum class TerminationReason {
+    Terminated,
+    ReachedCycle,
+    ReachedKnownCheckpoint,
+    MaxTapeLengthExceeded,
+    MaxEventCountExceeded,
+    TimeConstraintExceeded
+  };
+
+  struct EvaluationResult {
+    TerminationReason terminationReason;
+    uint64_t eventCount;
+    uint64_t maxTapeLength;
+    uint64_t finalTapeLength;
+  };
```

Comment:
It would be helpful for me if this struct included a copy of the original state, either as a `PostTagState`/`TagState` or preferably a `SmallState`.